hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/tagdetails/TagListField.java

```diff
@@ -0,0 +1,177 @@
+import org.eclipse.hawkbit.ui.utils.VaadinMessageSource;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Ordering;
+import com.google.common.collect.Sets;
+import com.vaadin.server.FontAwesome;
+import com.vaadin.ui.Button;
+import com.vaadin.ui.CssLayout;
+
+/**
+ * A panel that shows the assigned tags. A click on a tag unsassigns the tag
+ * from the {@link Target} or {@link DistributionSet}.
+ */
+public class TagListField extends CssLayout {
+
+    private static final long serialVersionUID = 1L;
+
+    private final transient Map<String, Button> tagButtons = new ConcurrentHashMap<>();
+    private final transient Set<TagAssignmentListener> listeners = Sets.newConcurrentHashSet();
+    private final VaadinMessageSource i18n;
+    private final boolean readOnlyMode;
+
+    /**
+     * Constructor.
+     * 
+     * @param i18n
+     * @param readOnlyMode
+     *            if <code>true</code> no unassignment can be done
+     */
+    TagListField(final VaadinMessageSource i18n, final boolean readOnlyMode) {
+        this.i18n = i18n;
+        this.readOnlyMode = readOnlyMode;
+        setSizeFull();
+    }
+
+    /**
+     * Initializes the Tag panel with all assigned tags.
+     * 
+     * @param assignedTags
+     *            assigned tags
+     */
+    void initializeAssignedTags(final List<TagData> assignedTags) {
+        removeAllComponents();
+
+        assignedTags.forEach(tag -> {
+            final Button tagButton = createButton(tag.getName(), tag.getColor());
+            tagButtons.put(tag.getName(), tagButton);
+        });
+
+        addTagButtonsAsComponents();
+    }
+
+    /**
+     * Adds a tag
+     * 
+     * @param tagName
+     * @param tagColor
+     */
+    void addTag(final String tagName, final String tagColor) {
+        if (!tagButtons.containsKey(tagName)) {
+            removeAllComponents();
+
+            final Button tagButton = createButton(tagName, tagColor);
+            tagButtons.put(tagName, tagButton);
+
+            addTagButtonsAsComponents();
+        }
+    }
+
+    private void addTagButtonsAsComponents() {
+        sortTagNames(tagButtons.keySet()).forEach(sortedTagName -> addComponent(tagButtons.get(sortedTagName)));
+    }
+
+    private static List<String> sortTagNames(final Iterable<String> tagNames) {
+        final List<String> sortedTagNames = Lists.newArrayList(tagNames);
+        sortedTagNames.sort(Ordering.natural());
+        return sortedTagNames;
+    }
+
+    private Button createButton(final String tagName, final String tagColor) {
+        final Button button = SPUIComponentProvider.getButton(UIComponentIdProvider.ASSIGNED_TAG_ID_PREFIX + tagName,
+                tagName, i18n.getMessage(UIMessageIdProvider.TOOLTIP_CLICK_TO_REMOVE), null, false, null,
+                SPUITagButtonStyle.class);
+        button.setData(tagName);
+        button.addClickListener(e -> removeTagAssignment((String) e.getButton().getData()));
```

Comment:
you don't need to use Buttons' data in here, just use `tagName` parameter directly